packages/langium/src/lsp/langium-lsp-module.ts

```diff
@@ -0,0 +1,115 @@
+
+/**
+ * Context required for creating the default language-specific dependency injection module.
+ */
+export interface LSPModuleContext {
+    shared: LangiumSharedLSPServices;
+}
+
+/**
+ * Create a dependency injection module for the default LSP services. This is a
+ * set of services that are used by exactly one language.
+ */
+export function createLSPModule(context: LSPModuleContext): Module<LangiumServices, LangiumLSPServices> {
+    return {
+        lsp: {
+            CompletionProvider: (services) => new DefaultCompletionProvider(services),
+            DocumentSymbolProvider: (services) => new DefaultDocumentSymbolProvider(services),
+            HoverProvider: (services) => new MultilineCommentHoverProvider(services),
+            FoldingRangeProvider: (services) => new DefaultFoldingRangeProvider(services),
+            ReferencesProvider: (services) => new DefaultReferencesProvider(services),
+            DefinitionProvider: (services) => new DefaultDefinitionProvider(services),
+            DocumentHighlightProvider: (services) => new DefaultDocumentHighlightProvider(services),
+            RenameProvider: (services) => new DefaultRenameProvider(services)
+        },
+        shared: () => context.shared
+    };
+}
+
+/**
+ * Create a dependency injection module for the LSP shared services. This is the set of
+ * services that are shared between multiple languages.
+ */
+export function createSharedLSPModule(context: DefaultSharedModuleContext): Module<LangiumSharedLSPServices & LangiumSharedCoreServices, LangiumSharedLSPServices> {
+    return {
+        // use the regular registry as an LSP registry, assuming the registry itself is opaque
+        ServiceRegistry: () => (new CoreServiceRegistry()) as ServiceRegistryI,
+        lsp: {
+            Connection: () => context.connection,
+            LanguageServer: (services) => new DefaultLanguageServer(services),
+            WorkspaceSymbolProvider: (services) => new DefaultWorkspaceSymbolProvider(services),
+            NodeKindProvider: () => new DefaultNodeKindProvider(),
+            FuzzyMatcher: () => new DefaultFuzzyMatcher()
+        },
+        workspace: {
+            TextDocuments: () => new TextDocuments(TextDocument),
+            WorkspaceManager: (services) => new LSPWorkspaceManager(services),
+            ConfigurationProvider: (services) => new LSPConfigurationProvider(services),
+            LangiumDocumentFactory: (services) => new LSPLangiumDocumentFactory(services),
+        }
+    };
+}
+
+/**
+ * Creates Langium grammar services, enriched with LSP functionality
+ *
+ * @param context Shared module context, used to create additional shared modules
+ * @param sharedModule Existing shared module to inject together with new shared services
+ * @returns Shared services enriched with LSP services + Grammar services, per usual
+ */
+export function createLangiumGrammarServicesWithLSP(context: DefaultSharedModuleContext,
```

Comment:
This should probably go into some `langium-grammar-module.ts` in `langium/grammar/lsp`, right?
conflicts with #1171...

Reply:
Good eyes, it's not quite where it should be yet. I'm waiting to rebase on #1171 , and am planning to build off of that. If it's already been moved there, I'll follow that.

Reply:
This has been removed after rebasing on the grammar export PR.